### PR TITLE
Add web form and users

### PR DESCRIPTION
**Pull request by drewpotter**
This is the initial work to add user account authentication and a form for editing:
- Events
- Organisers
- Categories

### CODE REVIEW

**Comment by rogeringle** [comment]
Would somebody be able to look at this so we can get this on staging for me to review?

We're keen to get this up and being used by Holly this week :)

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+before_script:
+- psql -c 'create database travis_ci_test' -U postgres
+- cp config/database.yml.travis config/database.yml
+- cp config/secrets.yml.travis config/secrets.yml
+- yarn install
+- RAILS_ENV=test bundle exec rake db:migrate
+- RAILS_ENV=test bundle exec rails test
+notifications:
+  slack:
+    secure: K7YDsuNqaWmKeoseHArxd/BIvGXoCCItbh/m6JLZWNTR4K/pogd4SrbC1SHDH7RmSc2otM+HAU3Xl85F4CE61XsMiZQ9Vpz1LslsuwStoS3DO62lejwTRpVINfJS535stbUWPX8bNiU2OxHl2BLB+e3yeNUkiO9mFA2TbcfXti0Su0qwF7XFbD6txEWrJ7s0Ix9hhDOfUyfVjjLMu5cVQUkqeT6jICN+zdA85qduBmiWRgz2KuhD4SYqz4Yrn/4LEZEnw/C54dlHM13p1/ZAO6ypfC6EUDT3T7s97CWiUHyby5zI7dMt2BR0P/P/SR5isg8v5OYDKri8/qKzzKC5jeejdYmuH8NlbMDnhdqKKAnbczljXVIfSyNDbuzB1DMu6ClvPm0sskOUH47Z7X2zLy4iYsPXA56vMU3WDGNnEG1N3V9Jw+phzSeYy5IV+P1kmsV8NRV9Ch000Cc8lyrhk8C1Me6CftUveZAxLaOca1TFya5eyBsw5YEecmx6RyKF38IbMazFfs/EkATHZxZE1EjZGKTTDtIjm0OTxQQAe43k8m6NJIOY6regZ8GoJagvp+TPftyKAQQkclymqYBxF6jkR4VKtgybAr2FWmu0NGgY6psGij7TFMz1yRutTIqeml2USlUjEUajHxVCLFnUuBPC6dEKtTIRuVFagxfIT/A=
```
Comment (Levia) [inline .travis.yml]: Are there any security concerns on having this key publicly available here?

Comment (drewpotter) [reply .travis.yml]: @Levia this is supposed to be the secure way of doing it, as far as I know from the docs 🤷‍♂ please advise if I have made a mistake here!

Comment (Levia) [reply .travis.yml]: That's cool! It looks like it's fine having it here, I've double checked

Comment (drewpotter) [reply .travis.yml]: Cool thanks!

**Comment by Levia** [comment]
@drewpotter Is there a way for the user/admin to be redirected to the editing section via the interface? At the moment it seems possible only if you know the url.

If this is meant to be like so, I'm wondering if there should be a more general landing page for admins rather than going straight away to events, organisers, etc.

**Comment by drewpotter** [comment]
@Levia thanks for pointing this out! Currently there is no mechanism for the user/admin to be redirected to the editing section via the interface. I was planning to provide the url to the events page - as I have done on staging. However, you are right to make this point. Perhaps we should have a general landing page?

**Comment by Levia** [comment]
@drewpotter If time allows perhaps it would be better. Otherwise we could leave it like this for now I guess.

**Comment by Levia** [comment]
@drewpotter Another thing I noticed is that none of the `destroy` methods are working for me. So I'm unable to delete events, organisers or categories

**Comment by drewpotter** [comment]
@Levia that is really strange - you are right - I used Rails scaffold for those and I have no idea why the `destroy` methods are not working. I will have to fix that!

**Comment by Levia** [comment]
@drewpotter @WKocur This looks ok to me and happy to merge.
There are just two things I would like to highlight for the next phase perhaps.

1. When creating a new event, the text inputs of the form seem to be inconsistent for me on Firefox in terms of font used. Some inputs show different fonts.
2. When creating an event with a given organiser and/or category, if I try to delete that category/organiser the system will let me, but not sure it should as the event created earlier will be then compromised. 
If an event can have no category and no organiser than I guess it's fine, but perhaps there should be a way to notify the user when there are linked events.

I will be happy to merge this anyway if you give me the ok.
@rogeringle FYI

**Comment by drewpotter** [comment]
Thanks @Levia ! I will just check with @rogeringle first

**Pull request merged**